requirements.txt

```diff
@@ -1,14 +1,29 @@
+# Core library dependencies
 CoolProp
-coverage
-coveralls
 jsonschema
-matplotlib
 numpy
-pandas
 pygfunction
 scipy
+
+###### AFTER THIS LINE ARE OPTIONAL DEVELOPER DEPENDENCIES ######
+######### KEEP THE LINE SAYING "OPTIONAL" IN PLACE HERE #########
+######### IT IS USED BY SETUP.PY WHEN BUILDING PACKAGES #########
+
+# OPTIONAL #
```

Comment:
Added a line in the requirements file that splits the required install deps from the optional deps.  In a future step, we could consider splitting the requirements into multiple files and just setting up dependencies between the files.  Or maybe this is easier solved with pyproject.toml.  *breathe*